### ISSUE TITLE
Add dry run option

### ISSUE THREAD

**Issue by willdavidow** [Equilibris/nx.nvim#18]
**Is your feature request related to a problem? Please describe.**
I'd like to understand the output of the `nx` command I'm running before I run it.

**Describe the solution you'd like**
It would be extremely useful to have an option to run a command using the `--dry-run` cli option, or perhaps a floating/split window that shows a dry run (similar to when changing an option in the VS Code plugin and watching the dry run happen in the terminal after each option change).

**Describe alternatives you've considered**
I tried hacking it into the plugin myself, but I'm too new to plugin development for Neovim to understand how/where the change should be made for this plugin and this particular feature.

**Additional context**
![Screenshot 2023-09-13 at 12 26 30 PM](https://github.com/Equilibris/nx.nvim/assets/763985/d78a2ccb-5a84-4487-9c46-fe51d35abb9f)

**Comment by Equilibris** [comment]
Hi as a quick fix ill add the dry-run option to all generators. adding the side panel is doable with telescope but i am a bit unsure how i would do it in the current framework since ive abstracted away telescope into 'form renderers' and 'code runners'. ill try to see what i can do tho

**Comment by willdavidow** [comment]
> Hi as a quick fix ill add the dry-run option to all generators. adding the side panel is doable with telescope but i am a bit unsure how i would do it in the current framework since ive abstracted away telescope into 'form renderers' and 'code runners'. ill try to see what i can do tho

Thanks! That's sort-of what I was trying to hack into the plugin by messing around with the `form_renderers.lua` file. I got the option in the form, but it ended up breaking because it wasn't part of the actual generator/form properties (or something like that) - still trying to wrap my head around how things piece together - and it ended up failing every time I tried to interact with the `dry run` boolean toggle I'd added

**Comment by Equilibris** [comment]
hi could you update the plugin and see if it works?

**Comment by willdavidow** [comment]
worked like a charm - thanks for the quick addition!
![image](https://github.com/Equilibris/nx.nvim/assets/763985/aa1ec064-f54a-4c98-8326-2c0cd10cce2e)

**Comment by Equilibris** [comment]
np! would you like to close the issue with this or should i try to expand the framework to allow for the sidebar idea?

**Comment by willdavidow** [comment]
I think it could be useful to include the sidebar, but if you don't have time (or desire) to look into that the change you've made here is great and works very well.

**Comment by Equilibris** [comment]
ill try to look at this but will be starting uni soon so wont have much time

**Comment by willdavidow** [comment]
Ok, no worries - like I said the option you added is great!

I don't have a ton of time either, but if you end up not having enough time I can also fork, try to add the functionality and submit a PR.

**Comment by Equilibris** [comment]
yeah, the plan i would go for to solve it (just writing it down in case i forget / if it helps) would be

- read up on telescopes previews
- add a 'preview' option for the form renderer, it could be useful if this could take a buffer so the command runner could simply be passed to 
- try to make a way to manage either the command runner or another module to produce the necessary preview for the plugin

as at least a baseline, dont know how much else would have to be done
